server/finatra-server/src/main/scala/tapir/server/finatra/FinatraRequestToRawBody.scala

```diff
@@ -0,0 +1,105 @@
+package tapir.server.finatra
+import java.io.{ByteArrayInputStream, File, FileOutputStream}
+import java.nio.ByteBuffer
+import java.nio.charset.Charset
+
+import com.twitter.finagle.http.Request
+import com.twitter.finatra.http.request.RequestUtils
+import com.twitter.io.Buf
+import org.apache.commons.fileupload.FileItemHeaders
+import tapir.model.Part
+import tapir.{
+  ByteArrayValueType,
+  ByteBufferValueType,
+  Defaults,
+  FileValueType,
+  InputStreamValueType,
+  MultipartValueType,
+  RawPart,
+  RawValueType,
+  StringValueType
+}
+
+import scala.collection.JavaConverters._
+
+object FinatraRequestToRawBody {
+  def apply[R](rawBodyType: RawValueType[R], body: Buf, charset: Option[Charset], request: Request): R = {
+
+    def asByteArray: Array[Byte] = {
+      val array = new Array[Byte](body.length)
+      body.write(array, 0)
+      array
+    }
+
+    def asByteBuffer: ByteBuffer = {
+      val buffer = ByteBuffer.allocate(body.length)
+      body.write(buffer)
+      buffer.flip()
+      buffer
+    }
+
+    rawBodyType match {
+      case StringValueType(defaultCharset) => new String(asByteArray, charset.getOrElse(defaultCharset))
+      case ByteArrayValueType              => asByteArray
+      case ByteBufferValueType             => asByteBuffer
+      case InputStreamValueType            => new ByteArrayInputStream(asByteArray)
+      case FileValueType                   => byteArrayToFile(asByteArray)
+      case mvt: MultipartValueType         => multiPartRequestToRawBody(request, mvt)
+    }
+  }
+
+  private def byteArrayToFile(byteArray: Array[Byte]): File = {
+    // TODO: Make this async
```

Comment:
akka/http4s interpreters contain an options object (eg AkkaHttpServerOptions), one of the fields there is `createFile: RequestContext => Future[File]`. Would it be possible to introduce one here, and make it return a (twitter) `Future[File]`?

Reply:
I didn't see the direct benefit of creating a ServerOptions object and just hardcoded everything because at this stage it didn't make sense to me why someone would override something -- perhaps that would be added later, I thought. 

However on the question of having a `RequestContext => Future[File]` -- the complicated part is not the ServerOptions, but that FinatraRequestToRawBody.apply would have to return a `Future`-- which is fine because it's inside code that returns a `Future` anyway... I deferred that change with that TODO, but can do it now if you'd like.

Reply:
being able to provide a custom file-creation function would be nice, as this allows to provide one which uses a dedicated execution context for blocking IO operations (file creation is such an operation). Plus it would be consistent with the rest :)